### PR TITLE
Update assembler to handle complex-valued equations

### PR DESCRIPTION
**Pull request by IgorBaratta**
- Allow  assembling of complex forms and complex BCs.
- Add new tests for complex assembling

The proposed tests are currently running in the complex branch, but they are skipped in the master branch.

### CODE REVIEW

File: cpp/dolfin/fem/FormIntegrals.h [hunk 1]
```diff
@@ -10,6 +10,7 @@
 #include <dolfin/common/types.h>
 #include <functional>
 #include <memory>
+#include <petscis.h>
```
Comment (garth-wells) [inline cpp/dolfin/fem/FormIntegrals.h]: Is the right header to include? Is it just to get `PetscScalar`?

File: cpp/dolfin/fem/LocalAssembler.h [hunk 2]
```diff
@@ -88,7 +88,7 @@ class LocalAssembler
...
       const mesh::MeshFunction<std::size_t>* interior_facet_domains, ///< [in]
       const mesh::MeshFunction<std::size_t>* cell_domains            ///< [in]
-      );
+  );
 };
-}
-}
\ No newline at end of file
+} // namespace fem
+} // namespace dolfin
```
Comment (garth-wells) [inline cpp/dolfin/fem/LocalAssembler.h]: Add new line.

File: cpp/dolfin/fem/SystemAssembler.h [hunk 3]
```diff
@@ -186,5 +187,5 @@ class SystemAssembler : public AssemblerBase
                        const std::vector<DirichletBC::Map>& boundary_values,
                        const common::ArrayView<const dolfin::la_index_t>& dofs);
 };
-}
-}
\ No newline at end of file
+} // namespace fem
+} // namespace dolfin
```
Comment (garth-wells) [inline cpp/dolfin/fem/SystemAssembler.h]: Add new line.

File: cpp/dolfin/fem/FormIntegrals.h (hunk 1 above)
Comment (IgorBaratta) [reply cpp/dolfin/fem/FormIntegrals.h]: No. It was meant to be `petscsys.h`. But in fact it's not really necessary.
We already get `PetscScalar` by importing `dolfin/common/types.h`.

File: cpp/dolfin/fem/LocalAssembler.h (hunk 2 above)
Comment (IgorBaratta) [reply cpp/dolfin/fem/LocalAssembler.h]: New line added at the end of the file.

**Pull request merged**